### ISSUE TITLE
[BUG]: Unable to connect to FreshRSS server (1.19.2)

### ISSUE THREAD

**Issue by gbakeman** [martinrotter/rssguard#743]
### Brief description of the issue

I'm not sure what I'm doing wrong. I've tried several different combinations of things, but cannot get a successful connection in RSS Guard 4.2.2 (installed from Chocolatey package manager). I've been able to connect to it just fine from Firefox desktop & mobile, as well as FocusReader for Android. I haven't seen anyone else post a bug like this, so maybe it's just me. I've attached the log but it doesn't appear to have anything helpful relating to the problem.

### How to reproduce the bug?

**Test 1**
URL: host.domain.org/freshrss/api/ (Recommended by FreshRSS)
User: [my user]
Password: [API password]

Result: Network error: 'content not found'

**Test 2**
URL: host.domain.org/freshrss/ (App root directory)
User: [my user]
Password: [API password]

Result: Network error: 'authentication faled'

**Test 3**
URL: host.domain.org/freshrss/ (App root directory)
User: [my user]
Password: [my password]

Result: Network error: 'authentication faled'

**Test 4**
URL: host.domain.org/freshrss/api/greader.php) (Recommended on /api/ page in browser)
User: [my user]
Password: [my password]

Result: Network error: 'authentication failed'

**Test 5**
URL: host.domain.org/freshrss/api/greader.php) (Recommended on /api/ page in browser)
User: [my user]
Password: [API password]

Result: Network error: 'authentication failed'

### What was the expected result?

A successful connection.

### What actually happened?

Different errors were returned depending on combination of input (see above)

### Debug log

time="     0.226" type="debug" -> core: Initializing settings in 'C:\Users\[user]\AppData\Local\RSS Guard 4\config\config.ini' (non-portable way).
time="     0.621" type="debug" -> database: File-based SQLite database connection 'DatabaseFactory' to file 'C:\Users\[user]\AppData\Local\RSS Guard 4\database\database.db' seems to be established.
time="     0.622" type="debug" -> database: File-based SQLite database has version '2'.
time="     0.624" type="debug" -> gui: Available icon theme paths: (:/icons, :/graphics, C:/Users/[user]/AppData/Local\RSS Guard 4\icons, C:/ProgramData/chocolatey/lib/rssguard/tools\icons)
time="     0.625" type="debug" -> gui: Installed icon themes are: '', 'Breeze', 'Breeze Dark', 'Faenza', 'Numix'
time="     0.627" type="debug" -> gui: Loading icon theme 'Breeze'.
time="     0.628" type="debug" -> gui: Found path of base skin: ':\skins\nudus-base'.
time="     0.628" type="debug" -> gui: Trying to load base file ':\skins\nudus-base\html_wrapper.html' for the skin.
time="     0.629" type="debug" -> gui: Local file ':\skins\nudus-light\html_style.css' exists, using it for the skin.
time="     0.629" type="debug" -> gui: Trying to load base file ':\skins\nudus-base\html_enclosure_image.html' for the skin.
time="     0.629" type="debug" -> gui: Trying to load base file ':\skins\nudus-base\html_single_message.html' for the skin.
time="     0.630" type="debug" -> gui: Trying to load base file ':\skins\nudus-base\html_enclosure_every.html' for the skin.
time="     0.630" type="debug" -> gui: Local file ':\skins\nudus-light\qt_style.qss' exists, using it for the skin.
time="     0.630" type="debug" -> gui: Trying to load base file ':\skins\nudus-base\html_adblocked.html' for the skin.
time="     0.631" type="debug" -> gui: Setting style: 'Fusion'.
time="     0.632" type="debug" -> gui: Skin 'nudus-light' loaded.
time="     0.632" type="debug" -> network: Persistent web data storage path: 'C:/Users/[user]/AppData/Local\RSS Guard 4\web\storage'.
time="     0.633" type="debug" -> network: Disabling application-wide proxy completely.
time="     0.637" type="debug" -> core: OpenSSL version: 'OpenSSL 1.1.1j  16 Feb 2021'.
time="     0.638" type="debug" -> core: OpenSSL supported: 'true'.
time="     0.638" type="debug" -> core: Starting RSS Guard 4.2.2.
time="     0.638" type="debug" -> core: Instantiated class  'Application'.
time="     0.639" type="debug" -> core: Starting to load active localization. Desired localization is 'en_US'.
time="     0.640" type="debug" -> core: Application localization 'en_US' loaded successfully, specifically sublocalization 'en_US' was loaded.
time="     0.640" type="warning" -> core: Qt localization 'en_US' WAS NOT loaded successfully.
time="     0.645" type="debug" -> database: SQLite database connection 'MessagesModel' to file 'C:/Users/[user]/AppData/Local/RSS Guard 4/database/database.db' seems to be established.
time="     0.649" type="debug" -> message-model: Repopulated model, SQL statement is now:
 'SELECT Messages.id, Messages.is_read, Messages.is_important, Messages.is_deleted, Messages.is_pdeleted, Messages.feed, Messages.title, Messages.url, Messages.author, Messages.date_created, Messages.contents, Messages.enclosures, Messages.score, Messages.account_id, Messages.custom_id, Messages.custom_hash, Feeds.title, CASE WHEN length(Messages.enclosures) > 10 THEN 'true' ELSE 'false' END AS has_enclosures FROM Messages LEFT JOIN Feeds ON Messages.feed = Feeds.custom_id AND Messages.account_id = Feeds.account_id WHERE 0 > 1;'.
time="     0.650" type="debug" -> core: Auto-download timer started with interval 60000 ms.
time="     0.650" type="debug" -> core: Creating FeedDownloader singleton.
time="     0.654" type="debug" -> gui: Creating main application form in thread: '0xc34'.
time="     0.665" type="debug" -> gui: Current row changed - proxy 'QModelIndex(-1,-1,0x0,QObject(0x0))', source 'QModelIndex(-1,-1,0x0,QObject(0x0))'.
time="     0.689" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="     0.754" type="debug" -> gui: Creating tray icon menu.
time="     0.791" type="debug" -> core: Showing the main window when the application is starting.
time="     0.863" type="debug" -> database: SQLite database connection 'FeedReader' to file 'C:/Users/[user]/AppData/Local/RSS Guard 4/database/database.db' seems to be established.
time="     0.865" type="debug" -> database: SQLite database connection 'FeedlyEntryPoint' to file 'C:/Users/[user]/AppData/Local/RSS Guard 4/database/database.db' seems to be established.
time="     0.867" type="debug" -> database: SQLite database connection 'GmailEntryPoint' to file 'C:/Users/[user]/AppData/Local/RSS Guard 4/database/database.db' seems to be established.
time="     0.868" type="debug" -> database: SQLite database connection 'GreaderEntryPoint' to file 'C:/Users/[user]/AppData/Local/RSS Guard 4/database/database.db' seems to be established.
time="     0.870" type="debug" -> database: SQLite database connection 'OwnCloudServiceEntryPoint' to file 'C:/Users/[user]/AppData/Local/RSS Guard 4/database/database.db' seems to be established.
time="     0.872" type="debug" -> database: SQLite database connection 'StandardServiceEntryPoint' to file 'C:/Users/[user]/AppData/Local/RSS Guard 4/database/database.db' seems to be established.
time="     0.873" type="debug" -> database: SQLite database connection 'TtRssServiceEntryPoint' to file 'C:/Users/[user]/AppData/Local/RSS Guard 4/database/database.db' seems to be established.
time="     0.874" type="debug" -> gui: User wants to have tray icon.
time="     0.875" type="debug" -> gui: Tray icon is available, showing now.
time="     0.878" type="debug" -> gui: Creating SystemTrayIcon instance.
time="     0.899" type="debug" -> gui: Showing tray icon immediately.
time="     0.963" type="debug" -> gui: Tray icon displayed.
time="     0.964" type="debug" -> core: No execution message received from other app instances.
time="     1.330" type="debug" -> gui: Hovered link: 'QUrl()'.
time="     1.636" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="     2.164" type="debug" -> network: Destroying Downloader instance.
time="     2.165" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="     4.673" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="     4.673" type="debug" -> oauth: User does not want handler to be running.
time="    33.538" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="    33.538" type="debug" -> oauth: User does not want handler to be running.
time="    33.540" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="    33.569" type="debug" -> network: Destroying Downloader instance.
time="    33.570" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="    33.573" type="debug" -> oauth: Destroying OAuth2Service instance.
time="    33.573" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="    60.656" type="debug" -> core: Starting auto-download event, remaining 15 minutes out of 15 total minutes to next global feed update.
time="   120.660" type="debug" -> core: Starting auto-download event, remaining 15 minutes out of 15 total minutes to next global feed update.
time="   122.414" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="   122.415" type="debug" -> oauth: User does not want handler to be running.
time="   122.416" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="   122.496" type="debug" -> network: Destroying Downloader instance.
time="   122.496" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="   122.498" type="debug" -> oauth: Destroying OAuth2Service instance.
time="   122.499" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="   176.359" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="   176.359" type="debug" -> oauth: User does not want handler to be running.
time="   176.360" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="   176.521" type="debug" -> network: Destroying Downloader instance.
time="   176.522" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="   176.524" type="debug" -> oauth: Destroying OAuth2Service instance.
time="   176.524" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="   180.653" type="debug" -> core: Starting auto-download event, remaining 15 minutes out of 15 total minutes to next global feed update.
time="   240.658" type="debug" -> core: Starting auto-download event, remaining 15 minutes out of 15 total minutes to next global feed update.
time="   293.422" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="   293.422" type="debug" -> oauth: User does not want handler to be running.
time="   293.424" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="   293.520" type="debug" -> network: Destroying Downloader instance.
time="   293.520" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="   293.523" type="debug" -> oauth: Destroying OAuth2Service instance.
time="   293.523" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="   300.657" type="debug" -> core: Starting auto-download event, remaining 15 minutes out of 15 total minutes to next global feed update.
time="   318.194" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="   318.195" type="debug" -> oauth: User does not want handler to be running.
time="   318.196" type="debug" -> network: Settings of BaseNetworkAccessManager loaded.
time="   318.292" type="debug" -> network: Destroying Downloader instance.
time="   318.294" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="   318.301" type="debug" -> oauth: Destroying OAuth2Service instance.
time="   318.302" type="debug" -> network: Destroying SilentNetworkAccessManager instance.
time="   360.656" type="debug" -> core: Starting auto-download event, remaining 15 minutes out of 15 total minutes to next global feed update.
time="   420.660" type="debug" -> core: Starting auto-download event, remaining 15 minutes out of 15 total minutes to next global feed update.
time="   432.228" type="debug" -> core: Cannot create new account.
time="   432.228" type="debug" -> gui: Destroying FormAddAccount instance.
time="   433.756" type="debug" -> gui: Main window's close event
time="   433.766" type="debug" -> gui: Main window's hide event
time="   435.279" type="debug" -> core: Cleaning up resources and saving application state.
time="   435.280" type="debug" -> core: Close lock was obtained safely.
time="   435.281" type="debug" -> feed-downloader: Destroying FeedDownloader instance.time="   435.282" type="debug" -> gui: Destroying FormMain instance.
time="   435.284" type="debug" -> gui: Destroying TabWidget instance.
time="   435.286" type="debug" -> gui: Destroying FeedMessageViewer instance.
time="   435.288" type="debug" -> gui: Destroying BaseToolBar instance.
time="   435.294" type="debug" -> gui: Destroying MessagesView instance.
time="   435.295" type="debug" -> gui: Destroying BaseToolBar instance.
time="   435.296" type="debug" -> gui: Destroying FeedsView instance.
time="   435.296" type="debug" -> gui: Destroying TabBar instance.
time="   435.297" type="debug" -> gui: Destroying StatusBar instance.
time="   435.299" type="debug" -> gui: Destroying SystemTrayIcon instance.
time="   435.349" type="debug" -> core: Destroying Application instance.
time="   435.350" type="debug" -> core: Destroying Mutex instance.
time="   435.493" type="debug" -> gui: Destroying IconFactory instance.
time="   435.494" type="debug" -> core: Destroying FeedReader instance.
time="   435.494" type="debug" -> feed-model: Destroying FeedsModel instance.
time="   435.495" type="debug" -> feed-model: Destroying FeedsProxyModel instance
time="   435.495" type="debug" -> message-model: Destroying MessagesModel instance.
time="   435.496" type="debug" -> message-model: Destroying MessagesProxyModel instance.

### Operating system and version

* OS: Windows 10 21H2
* RSS Guard version: 4.2.2

**Comment by gbakeman** [comment]
Just tried on devbuild f7fde856, with similar results. I also noticed how the URL field says "without any service-specific path" so I tried it with just host.domain.org, resulting in a 'content not found' error.

**Comment by martinrotter** [comment]
Hi, "without service-specific path" means to remove all service-specific parts of URL, like folders in service etc. But you must of course keep ROOT folder of your service (if you have any). How would otherwise RSS Guard know where your service runs?

Your "Test 2" case is correct. If RSS Guard reports your API pw/username as invalid, then it is invalid. Could you provide testing username/URL for me to test on your instance?

**Comment by gbakeman** [comment]
I just verified now that my username and API key are correct when receiving the authentication failed error. Giving you access will be tricky since this is a private self-hosted service that only I use (nothing is internet facing besides a VPN). For reference, my services are behind a Traefik proxy with a self-hosted PKI (self signed CA with signed certs). The CA cert is installed client-side so programs recognize my certificates.

Is there a dev build with more advanced logging I could run for you and provide results? Any HTTP/API commands you want me to run?

**Comment by martinrotter** [comment]
I see.

Yes, I will make adjustments and will print more logging messages in relevant places. Gimme some tome.

**Comment by martinrotter** [comment]
OK, I pushed 13a993434.

Wait until development build compiles and then test with it pls. Note that your username/password will get printed to log now, so perhaps use some artificial one-time API password etc.

https://github.com/martinrotter/rssguard/releases/tag/devbuild

Then post log here pls.

**Comment by martinrotter** [comment]
Ping.

**Comment by gbakeman** [comment]
Sorry, currently away from a PC but will get back next week.

**Comment by martinrotter** [comment]
ok. no prob, thanks for info.

**Comment by gbakeman** [comment]
@martinrotter Thank you for being so prompt and getting back to me with the devbuild. I've installed version 4.2.3 build 68c03688 from the devbuild section. It reports that the test login was successful, with the same credentials I used before. I reverted to 4.2.2 just to be sure, and it's working now too. I have to wonder if something changed on FreshRSS's side when I reset the API key?

Sorry about the false issue report. Again I really appreciate your help and how quickly you got back to me.
